### PR TITLE
Revamp Update package's unit tests (2/2)

### PR DESCRIPTION
**Pull request by stephanos**
## What changed?

- revamped the `update_test.go` tests from the ground up
- removed unnecessary `context.Context` parameters

## Why?

Ensure high test coverage (and remove duplicate tests).

## How did you test it?

There are no behavioral changes.

### CODE REVIEW

File: service/history/workflow/update/update.go [hunk 1]
```diff
@@ -595,15 +592,6 @@ func (u *Update) onResponseMsg(
...
 
-// isIncomplete checks if update is already completed (rejected or processed).
-func (u *Update) isIncomplete() bool {
-	return !u.state.Matches(stateSet(stateProvisionallyCompleted | stateCompleted | stateAborted))
-}
-
-func (u *Update) checkState(msg proto.Message, expected state) error {
-	return u.checkStateSet(msg, stateSet(expected))
-}
-
```
Comment (stephanos) [inline service/history/workflow/update/update.go:606]: unusued!

File: service/history/workflow/update/registry_test.go [hunk 2]
```diff
@@ -296,8 +285,8 @@ func TestFindOrCreate(t *testing.T) {
 		})
 
 		t.Run("sending 1st update still denies new update to be created", func(t *testing.T) {
-			_ = upd1.Send(
-				context.Background(), false, &protocolpb.Message_EventId{EventId: testSequencingEventID})
+			t.Helper()
```
Comment (alexshtin) [inline service/history/workflow/update/registry_test.go:288]: Is it really needed here?

File: service/history/workflow/update/update.go [hunk 3]
```diff
@@ -396,9 +397,8 @@ func (u *Update) needToSend(includeAlreadySent bool) bool {
 // If update is not in expected stateAdmitted, Send does nothing and returns nil.
 // If includeAlreadySent is set to true then Send will return message even if update was already sent but not processed by worker.
 // If update lacks a request then return nil; the request will be communicated to the worker via an UpdateAdmitted event.
-// Note: once update moved to stateSent it never moves back to stateRequested.
+// Note: once update moved to stateSent it never moves back to stateCreated.
```
Comment (alexshtin) [inline service/history/workflow/update/update.go:400]: It should be `stateAdmitted` here.

File: service/history/workflow/update/registry_test.go (hunk 2 above)
Comment (stephanos) [reply service/history/workflow/update/registry_test.go:288]: oh, it's not. I inlined a function here and didn't remove this one.

**Review by alexshtin — APPROVED** [review]

File: service/history/workflow/update/update.go (hunk 3 above)
Comment (stephanos) [reply service/history/workflow/update/update.go:400]: Right!

**Pull request merged**